core/src/main/java/hudson/Launcher.java

```diff
@@ -450,8 +450,14 @@ public ProcStarter writeStdin() {
 
         /**
          * Starts the new process as configured.
+         *
+         * @throws ArrayIndexOutOfBoundsException when {@link #cmds()} is empty
          */
         public Proc start() throws IOException {
+            if (cmds().isEmpty()) {
+                LOGGER.log(Level.WARNING, "Trying to start a process without setting a command. This might cause "
+                                       + "an ArrayIndexOutOfBounds exception");
+            }
```

Comment:
Is there a way to know what is going to launch the empty process. It would be great to have that information. If not, this message would not help to diagnose what is/will happening

Reply:
There is no command to launch, "".